### ISSUE TITLE
Failure to start Wallaby on cold start

### ISSUE THREAD

**Issue by AStoker** [wallabyjs/public#169]
It seems that whenever I do a reboot of my machine, when I open Visual Studio and start Wallaby, I always get this message during the stage of configuring Wallaby to run.

![screen shot 2015-06-23 at 12 09 36 pm](https://cloud.githubusercontent.com/assets/2907279/8366134/1842abba-1b60-11e5-9d92-1bcd4d4fa20e.png)

This causes Wallaby to hang, and I cannot start, stop, or restart Wallaby. The only way I can get it running is to restart Visual Studio, and then start Wallaby up again (which always works).  
So, to sum up, it seems that on first load of Visual Studio after booting up, Wallaby fails to start.

I'm running Windows 8.1, and running Visual Studio 2013. Any ideas on what to do?

**Comment by ArtemGovorov** [comment]
This is an interesting issue. From the symptoms, it looks like something in your system temporarily removing one of the wallaby.js files. 

Do you have any antivirus software installed? If so, could you please check if there's something related to the `server.js` file in its logs? 

Could you please try opening the `C:\Users\Andrew\.wallaby\core\` in Explorer right after your machine reboot (and before starting VS) and checking if the `server.js` file is there?

**Comment by ArtemGovorov** [comment]
Could you please also attach wallaby log file from `\Users\%username%\AppData\Local\Temp\.wallaby\` for the day when the issue happened?

**Comment by AStoker** [comment]
I have the default Windows Defender installed. Interesting problem indeed. 
There is nothing in the log regarding server.js.
The log file from the day of the problem says:
```
2015-06-25 18:53:11Z [31] Error Wallaby.AutoUpdater 0: Failed to get wallaby.json or update core component/extension.
System.IO.IOException: The directory is not empty.

   at System.IO.Directory.DeleteHelper(String fullPath, String userPath, Boolean recursive, Boolean throwOnTopLevelDirectoryNotFound)
   at System.IO.Directory.Delete(String fullPath, String userPath, Boolean recursive, Boolean checkHost)
   at System.IO.Directory.Delete(String path, Boolean recursive)  
   at .0aXO33KBrs4Br\]\,ZnS?"W6d.MoveNext() .  
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)  
   at )(txeNevoM.YWOQ:.\D:OOExdGhMuM_MMU?Q.
```
The file server.js exists, and the contents of the wallaby core directory are (before VS boot):
![screen shot 2015-06-26 at 9 51 28 am](https://cloud.githubusercontent.com/assets/2907279/8378688/2674d9b8-1be9-11e5-8ab6-072392fb9153.png)

For kicks and giggles, I decided to check out the folder contents of the wallaby core after I try to start wallaby, and at that time, only the ```runners``` folder exists (no server.js).

**Comment by ArtemGovorov** [comment]
@AStoker Thanks for the details, it's very helpful. Looks like the issue is in wallaby.js auto updater, I'll keep investigating.

**Comment by ArtemGovorov** [comment]
@AStoker Still can't reproduce it in my environment, but I have changed something that's maybe causing the issue. Please try the latest version (v1.0.11) and let me know how it goes.

> This causes Wallaby to hang, and I cannot start, stop, or restart Wallaby. 

Fixed. Now even if it hangs, Stop and Restart actions should be available.

**Comment by AStoker** [comment]
I'm pretty sure that whatever is breaking it is specific to my machine. However, after the update, the issue seems to be resolved. Not sure what was going on there. Thanks for taking the time to look into it!
